src/Connection/SocketCapturingConnection.hpp

```diff
@@ -0,0 +1,22 @@
+#ifndef PROXYTCP_SOCKETCAPTURINGCONNECTION_HPP
+#define PROXYTCP_SOCKETCAPTURINGCONNECTION_HPP
+#include "Connection.hpp"
+#include <fstream>
+
+namespace Proxy
+{
+    class Status;
+    class PCAPcaptringfile;
+    class SocketCapturingConnection: public Connection
+    {
+        class PCAPcapturingfile;
+    public:
+        SocketCapturingConnection();
```

Comment:
Unnecessary default constructor. SocketConnections shouldn't be able to be constructed by default constructor